docs/configuration.rst

```diff
@@ -0,0 +1,556 @@
+        "format": "<statsd message format>",
+        "tags": {
+            "<tag>": "<value>"
+        }
+    }
+
+  host
+    The statsd host address
+  port
+    The statsd listening port
+  format (default ``"telegraf"``)
+      Determines statsd message format. Following formats are supported:
+
+        - ``telegraf`` `Telegraf spec <https://github.com/influxdata/telegraf/tree/master/plugins/inputs/statsd>`_
+        - ``datadog`` `DataDog spec <http://docs.datadoghq.com/guides/dogstatsd/#datagram-format>`_
+
+  :tags: (default null)
+      The tag key can be used to enter optional tag values for the metrics
+push_gateway (default ``null``)
+  When set, enables sending metrics to a Prometheus Pushgateway with tags.
+  The value is a JSON obejct, for example::
+
+    {
+        "endpoint": "<pushgateway address>",
+        "tags": {
+            "<tag>": "<value>"
+        }
+    }
+
+  endpoint
+    The pushgateway address
+  tags
+    An object mapping tags to their values.
+
+
+.. _configuration_http:
+
+HTTP Server configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The pghoard daemon needs to listen on an HTTP port for the archive command and
+for fetching of basebackups/WAL's when restoring if not using an object store.
+FIXME: if it's not required, can we disable it ?
+
+http_address (default ``"127.0.0.1"``)
+               Address to bind the PGHoard HTTP server to.  Set to an empty string to
+               listen to all available addresses.
+http_port (default ``16000``)
+            HTTP webserver port. Used for the archive command and for fetching of
+            basebackups/WAL's when restoring if not using an object store.
+
+
+
+
+.. _configuration_compression:
+
+Compression
+~~~~~~~~~~~
+
+The PostgreSQL write-ahead log (WAL) and basebackups are compressed with
+Snappy (default), Zstandard (configurable, level 3 by default) or LZMA (configurable,
+level 0 by default) in order to ensure good compression speed and relatively small backup size.
+For performance critical applications it is recommended to test compression
+algorithms to find the most suitable trade-off for the particular use-case.
+E.g. Snappy is fast but yields larger compressed files, Zstandard (zstd) on the other hand
+offers a very wide range of compression/speed trade-off.
+
+The top-level ``compression`` key allows to define compression options::
+
+  {
+    "compression": {
+      "algorithm": "snappy",
+      "level": 3,
+      "thread_count": 4
+    }
+  }
+
+algorithm (default ``snappy``)
+            The compression algorithm to use. Available algorithms are
+            ``snappy``, ``zstd``, and ``lzma``
+level (default ``0`` for ``lzma`` and ``zstd``, ``3`` for ``snappy``)
+        The compression level to use. Depends on the algorithm used.
+thread_count (default to ``cpu_count`` + 1)
+        The number of threads used for parallel compression. FIXME: previous
+        doc wrote max(cpu_count, 5) but the code says otherwise
```

Comment:
This FIXME is confusing, I would just put the correct documentation, since it seems we know what is true.